### PR TITLE
An import step is not need a key to be defined.

### PR DESCRIPTION
**Pull request by towerhe**

### CODE REVIEW

File: test/test_data_miner.rb [hunk 1]
```diff
@@ -123,6 +123,37 @@
...
+        end
+      end
+    end
+    PetFunny.auto_upgrade!
 
+    before { PetFunny.delete_all }
+
+    it 'imports the example data' do
+      PetFunny.run_data_miner!
+      PetFunny.must_be :exists?
```
Comment (seamusabshere) [inline test/test_data_miner.rb:148]:
```
PetFunny.count.must_be :>, 0
```

perhaps?

File: lib/data_miner/step/import.rb [hunk 2]
```diff
@@ -115,6 +97,33 @@ def validate?
...
 
+      def upsert_enabled?
+        not validate? && (storing_primary_key? || table_has_autoincrementing_primary_key?)
+      end
+
+      def save_with_upsert
+        c = ActiveRecord::Base.connection_pool.checkout
+        Upsert.stream(c, model.table_name) do |upsert|
+          table.each do |row|
+            selector = { @key => attributes[@key].read(row) } if @key
```
Comment (seamusabshere) [inline lib/data_miner/step/import.rb]: how about

```
selector = @key ? { @key => attributes[@key].read(row) } : { model.primary_key => nil }
```

then below you get

```
upsert.row selector, document
```

Comment (towerhe) [inline lib/data_miner/step/import.rb]: +1, I will fix it.

File: test/test_data_miner.rb (hunk 1 above)
Comment (towerhe) [inline test/test_data_miner.rb:148]: IMO, They are the same.

**Comment by seamusabshere** [comment]
brilliant, thanks @towerhe !

**Pull request merged**